src/main/java/com/kobylchak/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,82 @@
+package com.kobylchak.bookstore.controller;
+
+import com.kobylchak.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.kobylchak.bookstore.dto.category.CategoryDto;
+import com.kobylchak.bookstore.dto.category.CreateCategoryRequestDto;
+import com.kobylchak.bookstore.service.book.BookService;
+import com.kobylchak.bookstore.service.category.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for managing categories")
+@SecurityRequirement(name = "bearerAuth")
+@RestController
+@RequestMapping("api/categories")
+@RequiredArgsConstructor
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ADMIN')")
+    public CategoryDto createCategory(@RequestBody @Valid CreateCategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    @Operation(summary = "Get all categories",
+               description = "API for getting list of all available categories")
+    public List<CategoryDto> getAll(Pageable pageable) {
```

Comment:
```suggestion
    public List<CategoryDto> getAllCategories(Pageable pageable) {
```
be consistent with namings